src/shared/filesystem.ts

```diff
@@ -24,9 +50,27 @@ export function mkdtempAsync(prefix: string): Promise<string> {
     })
 }
 
+export function readdirAsync(path: string | Buffer, options?: string | {}): Promise<string[]> {
```

Comment:
Returns `fs.Dirent[]` as well

https://nodejs.org/api/fs.html#fs_class_fs_dirent

When fs.readdir() or fs.readdirSync() is called with the `withFileTypes` option set to true, the resulting array is filled with `fs.Dirent` objects, rather than strings or Buffers.

Reply:
Several of the `fs` APIs (including this one) changed in NodeJS 10.x. We are on 8.11.2, which does not include the `Dirent[]` return type.

Reply:
Oops, my mistake then.